rcgen/src/key_pair.rs

```diff
@@ -595,6 +595,21 @@ impl TryFrom<&PrivateKeyDer<'_>> for KeyPair {
 	}
 }
 
+#[cfg(feature = "crypto")]
+impl From<KeyPair> for PrivatePkcs8KeyDer<'static> {
+	fn from(val: KeyPair) -> Self {
+		val.serialize_der().into()
+	}
+}
+
+#[cfg(feature = "crypto")]
+impl From<KeyPair> for PrivateKeyDer<'static> {
+	fn from(val: KeyPair) -> Self {
+		let der: PrivatePkcs8KeyDer = val.into();
+		der.into()
```

Comment:
This is a bit too much indirection for my taste. Does `Self::from(val.into())` work? Otherwise `Self::from(PrivatePkcs8KeyDer::from(val))`?

Reply:
Let me clarify.
1. We start with `KeyPair`.
2. `PrivateKeyDer` can be created from `PrivatePkcs8KeyDer`.
3. There are two `From` here to allow conversion to either one of the `Private*KeyDer`
4. The conversion to `PrivateKeyDer` defers to conversion to `PrivatePkcs8KeyDer` in order to avoid code duplication, and to ensure that both of them are covered by the single test.

Do you want me to just inline `From<KeyPair> for PrivatePkcs8KeyDer<'static>`
into `From<KeyPair> for PrivateKeyDer<'static>`,
or drop `From<KeyPair> for PrivatePkcs8KeyDer<'static>` altogether?

Reply:
I understand what it is doing. I just stylistically dislike type annotations, so I'd prefer to avoid the `into()` that needs a type annotation.